src/Slices/ServiceInstanceDetails/UI/Components/AttributesComponents/AttributesTable.tsx

```diff
@@ -0,0 +1,401 @@
+} from "@patternfly/react-core";
+import { EllipsisVIcon } from "@patternfly/react-icons";
+import {
+  Table,
+  Thead,
+  Tr,
+  Th,
+  TdProps,
+  TreeRowWrapper,
+  Td,
+  Tbody,
+  TableText,
+  ThProps,
+} from "@patternfly/react-table";
+import styled from "styled-components";
+import { InstanceAttributeModel } from "@/Core";
+import { words } from "@/UI";
+import {
+  AttributeSets,
+  formatTreeRowData,
+  sortTreeRows,
+  TreeRowData,
+} from "./Utils";
+
+interface Props {
+  dropdownOptions: string[];
+  attributeSets: Partial<Record<AttributeSets, InstanceAttributeModel>>;
+}
+
+/**
+ * The AttributesTable Component.
+ *
+ * This component is responsible of displaying the correct data in a TreeRowTable.
+ * It allows the user to select the attributeSet (Candidate, Active, Rollback) to display
+ * It has a dropdown menu to expand/collapse all rows.
+ *
+ * @Props {Props} - The props of the component
+ *  @prop {string[]} dropdownOptions - The dropdownOptions for the AttributeSetSelection
+ *  @prop {Partial<Record<AttributeSets, InstanceAttributeModel>>} attributeSets - The attributeSets available for the selected version.
+ * @returns {React.FC<Props>} A React Component displaying the Attributes in a TreeRowTable
+ */
+export const AttributesTable: React.FC<Props> = ({
+  dropdownOptions,
+  attributeSets,
+}) => {
+  const [selectedSet, setSelectedSet] = useState(dropdownOptions[0]);
+  const [expandedNodeIds, setExpandedNodeIds] = useState<string[]>([""]);
+
+  // Sort direction of the currently sorted column
+  const [activeSortDirection, setActiveSortDirection] = useState<
+    "asc" | "desc" | undefined
+  >();
+  const [activeSortIndex, setActiveSortIndex] = useState<number | undefined>();
+
+  // expandedDetailsNodeIds is used for the responsive view of the TreeTable.
+  const [expandedDetailsNodeIds, setExpandedDetailsNodeIds] = useState<
+    string[]
+  >([""]);
+  const [tableData, setTableData] = useState<TreeRowData[]>([]);
+  const [isToggleOpen, setIsToggleOpen] = useState(false);
+
+  /**
+   * Get the sorting parameters according to PF guidelines.
+   *
+   * @param {number} columnIndex - defaults to 1, since we only have one sortable header atm.
+   * @returns ThProps["sort"]
+   */
+  const getSortParams = (columnIndex: number = 1): ThProps["sort"] => ({
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+    },
+    onSort: (_event, index, direction) => {
+      setActiveSortIndex(index);
+      setActiveSortDirection(direction);
+    },
+    columnIndex,
+  });
+
+  /**
+   * Handles the change of the selected attribute Set.
+   *
+   * @param {React.FormEvent<HTMLSelectElement>} _event
+   * @param {string} value
+   */
+  const onSetSelectionChange = (
+    _event: React.FormEvent<HTMLSelectElement>,
+    value: string,
+  ) => {
+    setSelectedSet(value);
+  };
+
+  /**
+   * Handle the toggleClick event on the table option toggle.
+   */
+  const onToggleClick = () => {
```

Comment:
wouldn't it be better to have it in the toggle component as anonymous function, content is pretty straightforward, it's not reused so it should be fine.

Reply:
I suppose this is just preference to be honest, I tend to extract methods more easily, especially in components that can big. It has a difference on the memory usage, but in this case it's pretty trivial. using an anonymous function will be redeclared on each run, while an assigned one will not.

Reply:
Make sense!